widget/diagramwidget/forcelayout.go

```diff
@@ -0,0 +1,84 @@
+package diagramwidget
```

Comment:
I'm not sure how well this really works. I feel like when I left off it was still pretty rough. Maybe you patched it up a bit?

We may want to consider using `gonum/graph` for layout now or in the future https://pkg.go.dev/gonum.org/v1/gonum@v0.12.0/graph/layout

Perhaps it would make sense to define a `GraphLayouter` interface and refactor the force layout to implement it, that way folks can plug in their own layout engines without having to fork the library?

Reply:
I am not making any provisions for automatic layout. Diagram editors typically have multiple layout algorithms, but they are explicitly invoked by the user. Anyone building a diagram editor can add their favorite layout algorithms.